src/io/browser_http.ts

```diff
@@ -230,9 +232,20 @@ export class BrowserHTTPRequest implements IOHandler {
     return [
       weightSpecs,
       concatenateArrayBuffers(await loadWeightsAsArrayBuffer(
-          fetchURLs, this.requestInit, this.fetchFunc))
+          fetchURLs, this.requestInit, this.getFetchFunc()))
     ];
   }
+
+  /**
+   * Helper method to get the `fetch`-like function set for this instance.
+   *
+   * This is mainly for avoiding confusion with regard to what context
+   * the `fetch`-like function is bound to. In the default (browser) case,
+   * the function will be bound to `window`, instead of `this`.
+   */
+  private getFetchFunc() {
+    return this.fetchFunc || fetch;
```

Comment:
looks like this.fetchFunc is always set, not the getFetchFunc method is needed.

Reply:
yep. since you are using the getFetchFunc approach you can just skip binding fetch when assigning it to this.fetchFunc in the constructor. Unless you are also trying to protect against accidental direct usage of this.fetchFunc? In any case with the current setup the || part isn't needed.